src/main/perl/Application.pm

```diff
@@ -302,160 +292,141 @@ Initialize the Application.
-  }
+    my ($self, $command, @argv) = @_;
 
-  my %confvar=$self->{'CONFIG'}->varlist('^cfgfile$');
-  if (exists $confvar{'cfgfile'}) {
-    $configfile=$self->option("cfgfile") unless (defined $configfile);
+    $self->{'NAME'} = basename($command);
 
+    # who is the user
+    $self->{'USERNAME'} = (getpwuid($<))[0] || getlogin || undef;
+    # name of machine
+    my ($sysname,$nodename,$release,$version,$machine) = POSIX::uname();
+    $self->{'HOSTNAME'} = $nodename;
+
+    # instantiate default configuration options and values,
+    # as given by Application.pm and/or the Application,
+    # and pass them to AppConfig.
     #
-    if (defined $configfile) {
-      unless (-e $configfile) {
-        print STDERR "Warning: cannot read config file: $configfile, dropping.\n";
-      } else {
-        # read conf file options.
-        # Note that a 'cfgfile' option doesn't make
-        # any sense inside the cfgfile ;-)
-        unless ($self->{'CONFIG'}->file($configfile)) {
-          # exit if something is wrong according to AppConfig.
-          print STDERR "Problems parsing configuration file ".$configfile."\n";
-          exit(-1);
-          #        throw_error('problems reading configuration file: '.$configfile);
-          #        return undef;
+    # defaults: no option arguments,
+    #           expand internal vars in cf file, but scream if
+    #             an embedded variable is not defined (EXPAND_WARN)
+    #           option values are 'undef'
+    $self->{'CONFIG'}= AppConfig->new({
+        PEDANTIC => 1, # really needed?
+        CASE => 0,
+        GLOBAL => {
+            DEFAULT  => undef,
+            ARGCOUNT => ARGCOUNT_NONE,
+            EXPAND   => EXPAND_VAR|EXPAND_WARN
         }
-      }
+    });
+
+    # add application-specific options
+    unless ($self->_add_options()) {
+        throw_error('Cannot add options');
+        return undef;
     }
-  }
-  #
-  # if there was a request for --help, go for it before
-  # getting command line opt values (for getting the current
-  # default values before overwriting them with cmd line opts)
-  #
-  if ($help_request) {
-    $self->show_usage();
-    exit (0);
-  }
-  #
-  # now read in cmd line options, which always have highest priority.
-  # (uses Getopt::Long)
-  #
-  unless ($self->{'CONFIG'}->getopt(\@argv)) {
-    # exit if something is wrong according to AppConfig.
-    # print STDERR "problems reading command line\n";
-    exit(-1);
-    # throw_error('problems reading command line options');
-    # return undef;
-  }
 
-  #
-  # now we have all options available!
-  # Process now common options.
-  #
+    # check if we have a config file to read or a --help request.
+    # parse it 'by hand' instead of using AppConfig twice
+    # for cmd line parsing
+    my $help_request = 0;
+    my $configfile = undef;
+    my @args_tmp = @argv;
+    my $arg;
+
+    my $cfgfile_standalone_pattern = '^(--'.$OPTION_CFGFILE.')$';
```

Comment:
I am not convinced by this change from hardcoding the word to using a constant and then a variable. It's definitely not more readable than before, possibly harder to follow than before.